src/Helper/Helper_Notices.php

```diff
@@ -59,6 +59,20 @@ public function init() {
 	 */
 	public function add_actions() {
 		add_action( $this->get_notice_type(), [ $this, 'process' ] );
+		add_action( 'init', [ $this, 'maybe_remove_non_pdf_messages' ] );
+	}
+
+	/**
+	 * If the current page belongs to Gravity PDF, add these hooks.
+	 * Remove these hooks if the current page does not belong to Gravity PDF.
+	 *
+	 * @since 6.5
+	 */
+	public function maybe_remove_non_pdf_messages() {
+		if ( \GPDFAPI::get_misc_class()->is_gfpdf_page() ) {
```

Comment:
Let's reverse this condition and return early if not a GF page. It makes the flow easier to follow as you can have the main logic below it.